Sources/Peripheral/Peripheral.swift

```diff
@@ -279,25 +279,25 @@ extension Peripheral.DelegateWrapper: CBPeripheralDelegate {
     }
     
     func peripheral(_ cbPeripheral: CBPeripheral, didUpdateValueFor characteristic: CBCharacteristic, error: Error?) {
-        Task {
-            do {
-                let result = CallbackUtils.result(for: (), error: error)
-                try await self.context.readCharacteristicValueExecutor.setWorkCompletedForKey(
-                    characteristic.uuid, result: result
-                )
-            } catch {
-                if !characteristic.isNotifying {
+        
+        if characteristic.isNotifying {
+           
+           // characteristic.value is Data() and it will get trampled if allowed to run async.
+           self.context.characteristicValueUpdatedSubject.send( Characteristic(characteristic) )
+
+        } else {
```

Comment:
This will make it so that the read characteristic executor never properly dequeues `readValue` calls, since we're not calling `setWorkCompletedForKey` after completing the read value operation.
Could you describe the problem you're seeing a little more?

Reply:
Reading data from a gopro they have these complex packages where they split long packets into smaller ones. They will have to come in the same order as sent but it hits this race condition where the previous packet's data gets overwritten since its sending data by reference. maybe the data needs to be serialized per value then ?

Reply:
What's triggering the data to change?

Are you setting some value on the device, waiting for a notification on a characteristic and then reading its value (repeating this process until you've read all packets)?
If so, I wouldn't expect the values to change between reads, since you control when the next set of data comes in (e.g. you'd read the current data before writing to the characteristic to request for the next set of data).

If instead something else is changing the value of the characteristic you want to read (e.g. a timer on the gopro) and you're getting hit with two simultaneous `peripheral(_:didUpdateValueFor:error:)`, it might be a little trickier to solve. If you have some way of knowing the order of a given packet (e.g. continuation packet), one thing you could do is store the values in a "[continuation packet: data]" map and then sort the keys to read the data in order.

Reply:
Oh, I think there might be a simpler solution! Similar to what you've already done, but just allowing the read queue to continue:
```
if characteristic.isNotifying {
    self.context.characteristicValueUpdatedSubject.send(
        Characteristic(characteristic)
    )
}
Task {
    do {
        let result = CallbackUtils.result(for: (), error: error)
        try await self.context.readCharacteristicValueExecutor.setWorkCompletedForKey(
            characteristic.uuid, result: result
        )
    } catch {
        if !characteristic.isNotifying {
            Self.logger.warning("Received UpdateValue result for characteristic without a continuation")
        }
    }
}
```
The only drawback with this approach is that clients listening to the notification will get it before the read operation finishes. Which I think it's fine.

Reply:
So basically remove the else and let the Task go through either way? I could verify this and see if it works.
Thanks!

Reply:
Well, it seems to still work.

Reply:
Sick! I'll approve and merge!! 👍🏼